### PR TITLE
ttl: refactor TTL to seperate creating new job and take timeout job codes

### LINKED ISSUE #45285: refactor ttl to separate submit new job and take a timeout job codes

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #45285

There are two scenes for a tidb instance to lock and execute a job:

1. A table is time to execute, a new job should be created.
2. A exist job's heartbeat is timeout, the tidb should take over it.

In the previous implement this two scenes and processed in one method and not separated. However, in the future we want to move the trigger codes to timer so it's better to separate the above code from two scenes apart.

### What is changed and how it works?

Refactor method `lockNewJob` to `lockJob` and it can pass the extra argument `isCreate`, if `isCreate` is true, it only creates a new job and if `isCreate` is false, it only takes over the timeout job.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

File: ttl/ttlworker/job_manager.go [hunk 1]
```diff
@@ -498,25 +503,53 @@ tblLoop:
...
 }
 
-// couldTrySchedule returns whether a table should be tried to run TTL
-func (m *JobManager) couldTrySchedule(tableStatus *cache.TableStatus, table *cache.PhysicalTable, now time.Time, ignoreScheduleInterval bool) bool {
-	if tableStatus == nil {
-		// if the table status hasn't been created, return true
-		return true
-	}
+// couldLockJob returns whether a table should be tried to run TTL
+func (m *JobManager) couldLockJob(tableStatus *cache.TableStatus, table *cache.PhysicalTable, now time.Time, isCreate bool, checkScheduleInterval bool) bool {
```
Comment (bb7133) [inline ttl/ttlworker/job_manager.go:519]: How about splitting `couldLockJob` into something like `couldLockNewJob` and `couldLockExistingJob'?

Comment (bb7133) [reply ttl/ttlworker/job_manager.go:519]: `couldLockNewJob` and `couldLockExistingJob` share quite a few codes, and they will have slightly different arguments.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**